api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -361,4 +361,14 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
    *                                  change conflicts with other changes.
    */
   UpdateSchema moveAfter(String name, String afterName);
+
+
+  /**
+   * Applies all the additions and updates [type widening, field documentation]
+   * from the provided new schema
+   *
+   * @param newSchema - Input schema from which updates are applied
+   * @return this for method chaining
+   */
+  UpdateSchema unionWithByFieldName(Schema newSchema);
```

Comment:
@rdblue @rdsr pls weigh in on the naming of this new API - I went for something that's as accurate as possible but pls drop a note in case you have a better suggestion for it

Reply:
How about `unionByName` or `unionByNameWith`?

We'll also need to fill in the docs placeholder: `[type widening, field documentation]`.

Reply:
sure, I like either one @rdsr you make the call pls :)